android/src/main/java/io/ably/flutter/plugin/AblyMessageCodec.java

```diff
@@ -356,25 +369,31 @@ private TokenDetails decodeTokenDetails(Map<String, Object> jsonMap) {
 
   private ChannelOptions decodeRestChannelOptions(Map<String, Object> jsonMap) {
     if (jsonMap == null) return null;
-    final Object cipher = jsonMap.get(PlatformConstants.TxRealtimeChannelOptions.cipher);
-    try {
-      return createChannelOptions(cipher);
-    } catch (AblyException e) {
-      System.out.println("Exception while decoding RestChannelOptions: " + e);
-      return null;
+    ChannelOptions options = new ChannelOptions();
+    options.cipherParams = decodeCipherParams((Map<String, Object>) jsonMap.get(PlatformConstants.TxRestChannelOptions.cipherParams));
+    if (options.cipherParams != null) {
+      options.encrypted = true;
     }
+    return options;
+  }
+
+  /// RealtimeChannelOptions (the inheriting class) is used for both RealtimeChannelOptions and
```

Comment:
I'm not sure why this line starts with `///` and the next with `//`.

Reply:
9b42541c1d9a9aae5d83894d5e7d19a458361fdd